version 1.10.2 is broken: invalid memory address or nil pointer dereference

**.release** content:

```yaml
commitFormat: "angular"
branch:
  master: "release"
  develop: "rc"
release: "gitlab"
gitlab:
  repo: PROJECT-GROUP/sub-project-group/repository-name
  customUrl: https://git.my.org/
  tagPrefix: ""
```

build string:

```bash
$ CGO_ENABLED=0 go build -ldflags="-s -w" -a -installsuffix cgo -o build/go-semantic-release ./cmd/go-semantic-release/
```

comand to check:

```bash
$ ./go-semantic-release next
```

expected result:

```bash
$ ./go-semantic-release next
1.0.1-rc.0
```

actual result:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x14ff0da]

goroutine 1 [running]:
github.com/go-git/go-git/v5/plumbing.(*Reference).Hash(...)
	/Users/redacted/go/pkg/mod/github.com/go-git/go-git/v5@v5.2.0/plumbing/reference.go:183
github.com/Nightapes/go-semantic-release/internal/gitutil.(*GitUtil).GetCommits(0xc0000ba680, 0x0, 0x175cf3f, 0x31, 0xc0000ebb00, 0x1, 0x1)
	/Users/redacted/tmp/go-semantic-release-1.10.2/internal/gitutil/gitutil.go:126 +0x3a
github.com/Nightapes/go-semantic-release/pkg/semanticrelease.(*SemanticRelease).GetNextVersion(0xc0000d8820, 0xc00011c750, 0x0, 0x0, 0xc0000d8820, 0x0)
	/Users/redacted/tmp/go-semantic-release-1.10.2/pkg/semanticrelease/semantic-release.go:103 +0x138
github.com/Nightapes/go-semantic-release/cmd/go-semantic-release/commands.glob..func5(0x1b87ac0, 0x1bbd418, 0x0, 0x0, 0x0, 0x0)
	/Users/redacted/tmp/go-semantic-release-1.10.2/cmd/go-semantic-release/commands/next.go:52 +0x23d
github.com/spf13/cobra.(*Command).execute(0x1b87ac0, 0x1bbd418, 0x0, 0x0, 0x1b87ac0, 0x1bbd418)
	/Users/redacted/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:852 +0x472
github.com/spf13/cobra.(*Command).ExecuteC(0x1b87fc0, 0x1006045, 0xc00008c058, 0x101a501)
	/Users/redacted/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:960 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/redacted/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:897
github.com/Nightapes/go-semantic-release/cmd/go-semantic-release/commands.Execute(0x0, 0x0)
	/Users/redacted/tmp/go-semantic-release-1.10.2/cmd/go-semantic-release/commands/root.go:28 +0x55
main.main()
	/Users/redacted/tmp/go-semantic-release-1.10.2/cmd/go-semantic-release/main.go:10 +0x39
```

**note:** no issues for v1.10.1

--- a/internal/gitutil/gitutil.go
+++ b/internal/gitutil/gitutil.go
@@ -123,9 +123,18 @@ func (g *GitUtil) GetLastVersion() (*semver.Version, *plumbing.Reference, error)
 // GetCommits from git hash to HEAD
 func (g *GitUtil) GetCommits(lastTagHash *plumbing.Reference) ([]shared.Commit, error) {
 
-	excludeIter, err := g.Repository.Log(&git.LogOptions{From: lastTagHash.Hash()})
+	ref, err := g.Repository.Head()
 	if err != nil {
 		return nil, err
+	}
+	logOptions := &git.LogOptions{From: ref.Hash()}
+
+	if lastTagHash != nil {
+		logOptions = &git.LogOptions{From: lastTagHash.Hash()}
+	}
+	excludeIter, err := g.Repository.Log(logOptions)
+	if err != nil {
+		return nil, fmt.Errorf("could not get git log %w", err)
 	}
 
 	seen := map[plumbing.Hash]struct{}{}
@@ -142,10 +151,6 @@ func (g *GitUtil) GetCommits(lastTagHash *plumbing.Reference) ([]shared.Commit, 
 		return !ok && len(commit.ParentHashes) < 2
 	}
 
-	ref, err := g.Repository.Head()
-	if err != nil {
-		return nil, err
-	}
 	startCommit, err := g.Repository.CommitObject(ref.Hash())
 	if err != nil {
 		return nil, err

--- a/pkg/semanticrelease/semantic-release.go
+++ b/pkg/semanticrelease/semantic-release.go
@@ -1,6 +1,7 @@
 package semanticrelease
 
 import (
+	"fmt"
 	"github.com/Nightapes/go-semantic-release/internal/integrations"
 	"io/ioutil"
 	"time"
@@ -102,7 +103,7 @@ func (s *SemanticRelease) GetNextVersion(provider *ci.ProviderConfig, force bool
 
 	commits, err := s.gitUtil.GetCommits(lastVersionHash)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("could not get commits %w", err)
 	}
 
 	log.Debugf("Found %d commits till last release", len(commits))
@@ -131,11 +132,14 @@ func (s *SemanticRelease) GetNextVersion(provider *ci.ProviderConfig, force bool
 			Version: &newVersion,
 		},
 		Last: shared.ReleaseVersionEntry{
-			Commit:  lastVersionHash.Hash().String(),
+			Commit:  "",
 			Version: lastVersion,
 		},
 		Branch:  provider.Branch,
 		Commits: analyzedCommits,
+	}
+	if lastVersionHash != nil {
+		releaseVersion.Last.Commit = lastVersionHash.Hash().String()
 	}
 
 	if firstRelease {

